src/reader_native.cpp

```diff
@@ -83,22 +117,73 @@ int ReaderNative::read_time(bigint &ntimestep)
 
 void ReaderNative::skip()
 {
-  read_lines(2);
-  bigint natoms;
-  int rv = sscanf(line,BIGINT_FORMAT,&natoms);
-  if (rv != 1)
-    error->one(FLERR,"Dump file is incorrectly formatted");
-
-  read_lines(5);
-
-  // invoke read_lines() in chunks no larger than MAXSMALLINT
-
-  int nchunk;
-  bigint nremain = natoms;
-  while (nremain) {
-    nchunk = MIN(nremain,MAXSMALLINT);
-    read_lines(nchunk);
-    nremain -= nchunk;
+  if (binary) {
+    bigint natoms;
+    int triclinic;
+    skip_buf(sizeof(bigint));
+    read_buf(&triclinic, sizeof(int), 1);
+    skip_buf((sizeof(int)+sizeof(double))*6);
+    if (triclinic) {
+      skip_buf(sizeof(double)*3);
+    }
+    skip_buf(sizeof(int));
+
+    skip_reading_magic_str();
+
+    // read chunk and skip them
+
+    int nchunk;
+    read_buf(&nchunk, sizeof(int), 1);
+
+    int n;
+    for (int i = 0; i < nchunk; i++) {
+      read_buf(&n, sizeof(int), 1);
+      skip_buf(n*sizeof(double));
+    }
+
+    delete[] magic_string;
+    delete[] unit_style;
```

Comment:
if this is intentional, it should also add:
```suggestion
    delete[] unit_style;
    magic_string = nullptr;
    unit_style = nullptr;
```